src/app/utility/new-data-type-inline/new-data-type-inline.component.ts

```diff
@@ -188,4 +191,8 @@ export class NewDataTypeInlineComponent implements OnInit, AfterViewInit {
     this.validate();
     this.sendValidationStatus();
   };
+
+  toggleAdvanced(): void  {
+    this.advanced = !this.advanced;
```

Comment:
Could you update `onTypeSelect()` to set `advanced = false` please? I've identified an error where the `advanced` state is not reset correctly, as follows:

1. Change type from `Primitive` to `Reference`
2. Toggle the Advanced Options to see further fields
3. Now change type back to `Primitive`
4. And change back again to `Reference`. Note that the "Advanced options" toggle is in the off state, but the advanced options are visible. Now trying to toggle has the opposite effect.

You'll also need to bind the `advanced` field to the `mdm-slide-toggle` component to keep track of initial state correctly.

Reply:
Oh - yes, brilliant, thanks.  Much neater.